pepper-apis/src/main/java/org/broadinstitute/ddp/event/pubsubtask/api/PubSubTaskConnectionService.java

```diff
@@ -0,0 +1,139 @@
+ * PubSubTask 'taskType' (each processor implemented for a specific 'taskType').
+ * After doing an action it is called the {@link PubSubTaskResultSender}
+ * which sends result to topic with PubSubTask results.
+ */
+public class PubSubTaskConnectionService {
+
+    private static final Logger LOG = getLogger(PubSubTaskConnectionService.class);
+
+    public static final int DEFAULT_SUBSCRIBER_AWAIT_RUNNING_TIMEOUT_SEC = 30;
+    public static final int SUBSCRIBER_FAILURE_LISTENER_THREAD_COUNT = 4;
+    public static final int PUBLISHER_AWAIT_TERMINATION_TIMEOUT_MIN = 1;
+
+    private final PubSubConnectionManager pubSubConnectionManager;
+    private final ProjectSubscriptionName projectSubscriptionName;
+    private final ProjectTopicName pubSubTaskResultProjectTopicName;
+    private final PubSubTaskProcessorFactory pubSubTaskProcessorFactory;
+    private final int subscriberAwaitRunningTimeout;
+
+    private PubSubTaskReceiver pubSubTaskReceiver;
+    private PubSubTaskResultSender pubSubTaskResultSender;
+
+    private Subscriber pubSubTaskSubscriber;
+    private Publisher pubSubTaskResultPublisher;
+
+
+    public PubSubTaskConnectionService(
+            PubSubConnectionManager pubSubConnectionManager,
+            String projectId,
+            String subscription,
+            String pubSubTaskResultPubSubTopic,
+            int subscriberAwaitRunningTimeout,
+            PubSubTaskProcessorFactory pubSubTaskProcessorFactory) {
+        this.pubSubConnectionManager = pubSubConnectionManager;
+        projectSubscriptionName = ProjectSubscriptionName.of(projectId, subscription);
+        pubSubTaskResultProjectTopicName = ProjectTopicName.of(projectId, pubSubTaskResultPubSubTopic);
+        this.subscriberAwaitRunningTimeout = subscriberAwaitRunningTimeout;
+        this.pubSubTaskProcessorFactory = pubSubTaskProcessorFactory;
+    }
+
+    public void create() {
+        try {
+            createPubSubTaskResultPublisher();
+            pubSubTaskResultSender = new PubSubTaskResultSender(pubSubTaskResultPublisher);
+            pubSubTaskReceiver = new PubSubTaskReceiver(projectSubscriptionName, pubSubTaskProcessorFactory, pubSubTaskResultSender);
+            createPubSubTaskSubscriber();
+        } catch (Exception e) {
+            LOG.error(errorMsg("Failed to create PubSubTask connection"), e);
+        }
+    }
+
+    public void destroy() {
+        if (pubSubTaskSubscriber != null && pubSubTaskSubscriber.isRunning()) {
+            pubSubTaskSubscriber.stopAsync();
+        }
+        if (pubSubTaskResultPublisher != null) {
+            try {
+                pubSubTaskResultPublisher.shutdown();
+                pubSubTaskResultPublisher.awaitTermination(PUBLISHER_AWAIT_TERMINATION_TIMEOUT_MIN, TimeUnit.MINUTES);
+            } catch (Exception e) {
+                LOG.error(errorMsg("Failed to shutdown PubSubTask connection"), e);
```

Comment:
Seems like this error and the one on L76 are fatal--perhaps throw a runtime exception instead of logging?

Reply:
Ok. But seems it will be fatal for whole Housekeeping. Is it ok?

Reply:
How about wrapping this in a try/catch when it's called in `Housekeeping.java`?

Reply:
Fixed